### DISCUSSION TITLE
Please support pathlib.Path

### DISCUSSION THREAD

**Discussion by cdeil** [duckdb/duckdb#8878] [Ideas]
Please support [pathlib.Path](https://docs.python.org/3/library/pathlib.html) in duckdb!

It was added to the Python standard library 10 years ago and has become standard.

See https://github.com/duckdb/duckdb/issues/8182
cc @ned2

**Comment by Mjboothaus** [comment]
Same as https://github.com/duckdb/duckdb/discussions/6447 maybe (or least part of it).

I think it is a worthwhile and I imagine relatively small fix? (Maybe I'm too optimistic!) Luv the Duck!

**Reply by Tishj** [reply]
I think that makes sense.

Similar to https://github.com/duckdb/duckdb/pull/6568 we would just convert the `pathlib.Path` object into a string
Do we just convert it with `str` or should special care be taken?

**Reply by soerenwolfers** [reply]
@Tishj you should use `str`. The [ source](https://github.com/python/cpython/blob/d5df25268b037c2f1f4a1184fad5274d0e72f8d6/Lib/pathlib/_abc.py#L164) of `as_posix` is 
```python
def as_posix(self):
        """Return the string representation of the path with forward (/)
        slashes."""
        return str(self).replace(self.parser.sep, '/')
```

**Comment by Mjboothaus** [comment]
I typically use str(path_variable_name) or path_variable_name.as_posix()
Whatever works best for you :)

**Comment by Tishj** [comment]
This is now supported in `connect` with <https://github.com/duckdb/duckdb/pull/12809>

As mentioned there, this should serve as a blueprint for the implementation in some other methods that might benefit from accepting pathlib.Path objects

**Comment by Mjboothaus** [comment]
Thanks! :)
